sdk/tables/azure-data-tables/azure/data/tables/_table_service_client.py

```diff
@@ -148,13 +141,11 @@ def set_service_properties(self, **kwargs) -> None:
         :raises: :class:`~azure.core.exceptions.HttpResponseError`
         """
         cors = kwargs.pop("cors", None)
-        if cors:
-            cors = [c._to_generated() for c in cors]  # pylint:disable=protected-access
         props = TableServiceProperties(
             logging=kwargs.pop("analytics_logging", None),
             hour_metrics=kwargs.pop("hour_metrics", None),
             minute_metrics=kwargs.pop("minute_metrics", None),
-            cors=cors,  # type: ignore
+            cors=[c._to_generated() for c in cors] if cors is not None else cors,  # pylint:disable=protected-access
```

Comment:
I'm not sure if the explicit None-check is necessary?
```suggestion
            cors=[c._to_generated() for c in cors] if cors else cors,  # pylint:disable=protected-access
```

Reply:
It's not necessary, updated

Reply:
Update: it's not necessary in sync bit necessary in async, because `cors` type is specified in async api, without None check will have this mypy error:
```
error: Argument "cors" to "TableServiceProperties" has incompatible type "Optional[List[TableCorsRule]]"; expected "Optional[List[CorsRule]]"  [arg-type]
```

Reply:
There should be no difference between them - sounds like we're doing some weird in one of them?